Mathlib/LinearAlgebra/Matrix/ToLin.lean

```diff
@@ -0,0 +1,980 @@
+#align matrix.to_lin_alg_equiv_one Matrix.toLinAlgEquiv_one
+
+theorem LinearMap.toMatrixAlgEquiv_reindexRange [DecidableEq M₁] (f : M₁ →ₗ[R] M₁) (k i : n) :
+    LinearMap.toMatrixAlgEquiv v₁.reindexRange f ⟨v₁ k, mem_range_self k⟩ ⟨v₁ i, mem_range_self i⟩ =
+      LinearMap.toMatrixAlgEquiv v₁ f k i :=
+  by simp_rw [LinearMap.toMatrixAlgEquiv_apply, Basis.reindexRange_self, Basis.reindexRange_repr]
+#align linear_map.to_matrix_alg_equiv_reindex_range LinearMap.toMatrixAlgEquiv_reindexRange
+
+theorem LinearMap.toMatrixAlgEquiv_comp (f g : M₁ →ₗ[R] M₁) :
+    LinearMap.toMatrixAlgEquiv v₁ (f.comp g) =
+      LinearMap.toMatrixAlgEquiv v₁ f ⬝ LinearMap.toMatrixAlgEquiv v₁ g :=
+  by simp [LinearMap.toMatrixAlgEquiv, LinearMap.toMatrix_comp v₁ v₁ v₁ f g]
+#align linear_map.to_matrix_alg_equiv_comp LinearMap.toMatrixAlgEquiv_comp
+
+theorem LinearMap.toMatrixAlgEquiv_mul (f g : M₁ →ₗ[R] M₁) :
+    LinearMap.toMatrixAlgEquiv v₁ (f * g) =
+      LinearMap.toMatrixAlgEquiv v₁ f ⬝ LinearMap.toMatrixAlgEquiv v₁ g := by
+  rw [LinearMap.mul_eq_comp, LinearMap.toMatrixAlgEquiv_comp v₁ f g]
+#align linear_map.to_matrix_alg_equiv_mul LinearMap.toMatrixAlgEquiv_mul
+
+theorem Matrix.toLinAlgEquiv_mul (A B : Matrix n n R) :
+    Matrix.toLinAlgEquiv v₁ (A ⬝ B) =
+      (Matrix.toLinAlgEquiv v₁ A).comp (Matrix.toLinAlgEquiv v₁ B) :=
+  by convert Matrix.toLin_mul v₁ v₁ v₁ A B
+#align matrix.to_lin_alg_equiv_mul Matrix.toLinAlgEquiv_mul
+
+@[simp]
+theorem Matrix.toLin_finTwoProd_apply (a b c d : R) (x : R × R) :
+    Matrix.toLin (Basis.finTwoProd R) (Basis.finTwoProd R) !![a, b; c, d] x =
+      (a * x.fst + b * x.snd, c * x.fst + d * x.snd) :=
+  by simp [Matrix.toLin_apply, Matrix.mulVec, Matrix.dotProduct]
+#align matrix.to_lin_fin_two_prod_apply Matrix.toLin_finTwoProd_apply
+
+theorem Matrix.toLin_finTwoProd (a b c d : R) :
+    Matrix.toLin (Basis.finTwoProd R) (Basis.finTwoProd R) !![a, b; c, d] =
+      (a • LinearMap.fst R R R + b • LinearMap.snd R R R).prod
+        (c • LinearMap.fst R R R + d • LinearMap.snd R R R) :=
+  LinearMap.ext <| Matrix.toLin_finTwoProd_apply _ _ _ _
+#align matrix.to_lin_fin_two_prod Matrix.toLin_finTwoProd
+
+@[simp]
+theorem toMatrix_distrib_mul_action_toLinearMap (x : R) :
+    LinearMap.toMatrix v₁ v₁ (DistribMulAction.toLinearMap R M₁ x) = Matrix.diagonal fun _ => x :=
+  by
+  ext
+  rw [LinearMap.toMatrix_apply, DistribMulAction.toLinearMap_apply, LinearEquiv.map_smul,
+    Basis.repr_self, Finsupp.smul_single_one, Finsupp.single_eq_pi_single, Matrix.diagonal_apply,
+    Pi.single_apply]
+#align to_matrix_distrib_mul_action_to_linear_map toMatrix_distrib_mul_action_toLinearMap
+
+end ToMatrix
+
+namespace Algebra
+
+section Lmul
+
+variable {R S : Type _} [CommRing R] [Ring S] [Algebra R S]
+
+variable {m : Type _} [Fintype m] [DecidableEq m] (b : Basis m R S)
+
+theorem toMatrix_lmul' (x : S) (i j) :
+    LinearMap.toMatrix b b (LinearMap.Algebra.lmul R S x) i j = b.repr (x * b j) i := by
```

Comment:
`LinearMap.Algebra.lmul` shouldn't exist; #3630 fixes the name

Reply:
@Ruben-VandeVelde; we'll need to rebase again after the next (two-hourly?) mathport run in order to get this fix.